src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -583,3 +608,63 @@ def test_fuse_scalars_incorrect(parser):
         fuse_loops(parser, code)
     assert "Scalar variable 'b' might not be written in one loop" \
         in str(err.value)
+
+
+# ----------------------------------------------------------------------------
+def test_fuse_no_symbol(parser, monkeypatch):
+    '''Tests what happens if a variable name is not in the symbol table,
+    or not fully defined (i.e. likely imported from another module).
+    We have to patch the object (to replace the existing symbol table)
+    since otherwise all variables are in the symbol table.
+    '''
+    # Case 1: assume that the array 't' is imported from mymod. In
+    # this case the loop validation will find a Symbol (not a DataSymbol),
+    # and cannot test if this variable is an array. It should fall back
+    # to use the variable accesses information (which includes indices),
+    # knowing this way that this is an array.
+    code = '''subroutine sub()
+              use mymod
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s
+              do jj=1, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=1, n
+                 do ji=1, 10
+                    t(ji, jj) = s(ji, jj) + t(ji, jj)
+                 enddo
+              enddo
+              end subroutine sub'''
+    out, psyir = fuse_loops(parser, code)
+    assert """
+  do jj = 1, n, 1
+    do ji = 1, 10, 1
+      s(ji,jj) = t(ji,jj) + 1
+    enddo
+    do ji = 1, 10, 1
+      t(ji,jj) = s(ji,jj) + t(ji,jj)
+    enddo
+  enddo""" in out
+
+    # Case 2: remove the symbol table, which means that not
+    # even a Symbol instance is returned. This should then look
+    # up the information in the variable access information
+    # to determine which variables are an array
+    loop = psyir.children[0]
+    new_symbol_table = SymbolTable()
```

Comment:
I can imagine that this might cause problems in the future (as the SymbolTable becomes more integrated). It might be better to be more 'surgical' and just remove the `ContainerSymbol` "mymod" from the SymbolTable - if that works?

Reply:
I thought I had tried that, and got an exception that this type of symbol could not be removed ... I tried again, and it works now (oh, and I need to remove the  symbol 't', not 'mymod' - this way 't' is not found in the symbol table, which then triggers the use of the 'symbol not in symbol table' branch in the code).